sphinx.go

```diff
@@ -187,7 +187,7 @@ func generateSharedSecrets(paymentPath []*btcec.PublicKey,
 		)
 	}
 
-	return hopSharedSecrets, nil
+	return hopSharedSecrets, lastEphemeralPubKey, nil
```

Comment:
Perhaps time to make into a struct return value? 

Would any of the other intermediate values be useful here? Maybe for doing self-processing of non dummy hops in a blinded path?

Reply:
the only other thing we need to do the self-processing is the session key which we anyways pass in to the package from the above layer. 

But yeah - happy to turn this into a struct if better!

Reply:
updated the exported `BuildBlindedPath` function to return a new `BlindedPathInfo` struct